### PR TITLE
Added Apache Spark - What does going from 2.4 to 3.5 get you?

### PR DESCRIPTION
**Pull request by sconway-scottlogic**
[https://sconway-scottlogic.github.io/blog/2024/04/22/going-from-2.4-to-3.5.html](https://sconway-scottlogic.github.io/blog/2024/04/22/going-from-2.4-to-3.5.html)

Have you (please tick each box to show completion):

 - [x] Added your blog post to a single category?
 - [x] Added a brief summary for your post? Summaries should be roughly two sentences in length and give potential readers a good idea of the contents of your post.
 - [ ] Checked that the build passes?
 - [x] Checked your spelling (you can use `npm spellcheck` if that's your thing)
 - [x] Ensured that your author profile contains a profile image, and a brief description of yourself? (make it more interesting than just your job title!)
 - [x] Optimised any images in your post? They should be less than 100KBytes as a general guide.

Posts are reviewed / approved by your Regional Tech Lead.

### CODE REVIEW

File: _posts/2024-04-22-going-from-2.4-to-3.5.md [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+([from the official documentation found here](https://spark.apache.org/spark-connect/))
+
+- A connection is established between the Client and Spark Server
+- The Client converts a DataFrame query to an unresolved logical plan that describes the
+  intent of the operation rather than how it should be executed
+- The unresolved logical plan is encoded and sent to the Spark Server
+- The Spark Server optimises and executes the query
+- The Spark Server sends the results back to the Client
+
+spark-connect-communication.png
```
Comment (jwalker-scottlogic) [inline _posts/2024-04-22-going-from-2.4-to-3.5.md]: Looks like this line has snuck in by mistake?

File: _posts/2024-04-22-going-from-2.4-to-3.5.md [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+  </tbody>
+</table>
+
+# To Conclude
+
+Migrating software to newer versions is always daunting, but moving on from out of
+support versions is a crucial process.
+
+As seen above, there are relatively few deprecations. A migration guide is provided
+here, and includes a number of configuration switches to retain legacy behaviours,
```
Comment (jwalker-scottlogic) [inline _posts/2024-04-22-going-from-2.4-to-3.5.md]: Is there a missing link on the `here` ?

File: _posts/2024-04-22-going-from-2.4-to-3.5.md (hunk 1 above)
Comment (sconway-scottlogic) [reply _posts/2024-04-22-going-from-2.4-to-3.5.md]: Yes indeed. Fixed now.

File: _posts/2024-04-22-going-from-2.4-to-3.5.md (hunk 2 above)
Comment (sconway-scottlogic) [reply _posts/2024-04-22-going-from-2.4-to-3.5.md]: Good spot. I have added the missing link and reworded slightly.

**Pull request merged**